docs/blog/2018-11-06-gatsby-for-apps/index.md

```diff
@@ -0,0 +1,212 @@
+
+To begin... what even is an app, anyways?
+
+## What is an app?
+
+I've gone into the [ins and outs of describing the difficulty][whats-an-app] of defining a traditional web application. In an effort to not re-hash all the work there, I think there are a few simple features that, when present, indicate a more app-like experience. Specifically:
+
+- dynamic data fetching
+- user authentication and authenticated client-only routes
+- client-side JS interactions
+
+Of course, a web app isn't some checklist wherein _each_ of these functionalities are required to indicate an app-like experience. Rather, I think it's easier to _see_ an example of a web application to form a mental model of the type of web app that Gatsby can build.
+
+For me, there are two, key examples that are at the forefront of my mental model of what constitutes a modern web application... Gmail and Twitter (Lite).
+
+### Gmail
+
+![Gmail interface](./images/gmail.png)
+
+Gmail served as an early proof of concept that proved out two, key functional wins:
+
+1. Client-side JavaScript can power an app-like experience, and
+2. a JavaScript application (running in your browser) can compare favorably to traditional, native applications for desktop and mobile
+
+The impact of these wins can’t be understated. It _proved_ that a native, app-like experience is not only possible for end users, but that it can be
+
+Next up, is Twitter, for slightly different reasons.
+
+### Twitter (Lite / PWA)
+
+![Twitter interface](./images/twitter.png)
+
+Twitter is another key example of my mental model for what an application _is_ (and can be!) because it both:
+
+- illustrates some of the power of the modern web experience, and
+- uses some great performance optimizations and smart engineering practices to serve up an engaging, fast, app-like experience
+
+In particular, I find the following functionalities of Twitter Lite as key determinants for a separate kind of application:
+
+- Aggressive data caching and fast page navigation with service workers
+- Integration of the [PRPL pattern][prpl] (**P**ush, **R**ender, **P**re-cache, and **L**azy load)
+- The illustration of the [App Shell][app-shell] pattern to speed up repeat visits and show a maximally visually complete page
+
+These _modern_ concepts, coupled together, serve as a key marker for the value of Twitter Lite's approach to an app-like experience. They were able to strip down the core Twitter experience and deliver a blazing-fast modern web application utilizing some great engineering techniques and patterns. To learn more about these techniques, check out this great [case study][case-study].
+
+These two, great applications will serve as key foundational pieces to keep in mind as the discussion now slightly pivots to Gatsby for web applications.
+
+## Gatsby for Apps
```

Comment:
I was expecting to hear a clear explanation here of how Gatsby can build things that are like Gmail and Twitter Lite, and then got lost and couldn't find what I was expecting.